core/trie/src/lib.rs

```diff
@@ -137,9 +137,23 @@ where
 	)
 }
 
-/// Determine whether a child trie key is valid. `child_trie_root` and `child_delta_trie_root` can panic if invalid value is provided to them.
-pub fn is_child_trie_key_valid<H: Hasher>(_storage_key: &[u8]) -> bool {
-	true
+/// Determine whether a child trie key is valid.
+///
+/// For now, the only valid child trie key is `:child_storage:default:`.
+///
+/// `child_trie_root` and `child_delta_trie_root` can panic if invalid value is provided to them.
+pub fn is_child_trie_key_valid<H: Hasher>(storage_key: &[u8]) -> bool {
+	use substrate_primitives::storage::well_known_keys;
+	let has_right_prefix = storage_key.starts_with(b":child_storage:default:");
```

Comment:
this can probably be changed to a `well_known_keys::CHILD_... .iter().chain(":default).iter()).to_vec()` that will get optimised at compile time.
Though I'm not sure so we could wait for const fn to be sure.

Reply:
I am afraid that it wouldn't be the case. AFAIU, it would be hard for LLVM to optimize vector usage like this.

Reply:
[Here](https://godbolt.org/z/mm0IYr) is what compiler explorer shows